presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -283,6 +284,11 @@ public SystemSessionProperties(
                         featuresConfig.isLegacyOrderBy(),
                         false),
                 booleanSessionProperty(
+                        SUPPORT_NULL_ON_LHS_OF_SEMI_JOIN,
```

Comment:
Anyway, I dropped the feature toggle and updated implementation accordingly (hence rebase...). I also added a TODO in `EffectivePredicateExtractor` where it is incorrect (projections and nulls). There was an existing TODO, but it was too brief to be clear.

Maybe fixing `EffectivePredicateExtractor` is what it takes to re-enable certain pushdowns.

@martint  back to you.